stablehlo/reference/Ops.cpp

```diff
@@ -850,6 +862,61 @@ Tensor evalFloorOp(const Tensor &operand, ShapedType resultType) {
   return result;
 }
 
+Tensor evalGatherOp(const Tensor &operand, const Tensor &startIndices,
+                    const Axes &offsetDims, const Axes &collapsedSliceDims,
+                    const Axes &startIndexMap, Axis indexVectorDim,
+                    const Sizes &sliceSizes, bool indicesAreSorted,
+                    ShapedType resultType) {
+  Tensor result(resultType);
+  Axes batchDims;
+  for (auto i = 0; i < result.getRank(); ++i)
+    if (!llvm::is_contained(offsetDims, i)) batchDims.push_back(i);
+
+  for (auto resultIt = result.index_begin(); resultIt != result.index_end();
+       ++resultIt) {
+    Index batchIndex;
+    for (auto batchDim : batchDims) batchIndex.push_back((*resultIt)[batchDim]);
+
+    Index sliceStartIndices(batchIndex);
+    Index sliceLimitIndices(Sizes(sliceStartIndices) + 1);
```

Comment:
why `+ 1` ? It should the same batch index as sliceStartIndices?

Reply:
The slice index is from [startIndex, limitIndex). Without +1, it would be [batchIndex, batchIndex) which would be nothing.